src/api/serializers.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework.serializers import ModelSerializer
+
+from account.models import Customer
+from music.models import Album, Artist, Genre, Playlist, Track
+
+
+class CustomerSerializer(ModelSerializer):
+    class Meta:
+        model = Customer
+        fields = ("first_name", "last_name", "email", "password", "is_staff")
+
+
+class ArtistSerializer(ModelSerializer):
+    class Meta:
+        model = Artist
+        fields = ("id", "name")
+
+
+class AlbumSerializer(ModelSerializer):
+    class Meta:
+        model = Album
+        fields = ("id", "title")
+
+
+class TrackSerializer(ModelSerializer):
+    artist = ArtistSerializer()
+    album = AlbumSerializer()
+
+    class Meta:
+        model = Track
+        fields = ("id", "title", "length", "artist", "album")
+
+    def create(self, validated_data):
+        artist_data = validated_data.pop("artist")
```

Comment:
Add validation
In the TrackSerializer's create() method, add validation to ensure that both the artist and album fields are present in the incoming data.

 if not artist_data or not album_data:
            raise ValueError("Both artist and album fields are required.")